rio_tiler/reader.py

```diff
@@ -206,6 +215,34 @@ def part(
 
     out_height = height or vrt_height
     out_width = width or vrt_width
+
+    if buffer:
+        height = height or vrt_height
+        width = width or vrt_width
+
+        # Get output resolution
+        x_res = (bounds[2] - bounds[0]) / out_width
+        y_res = (bounds[3] - bounds[1]) / out_height
+
+        # apply buffer to bounds
+        bounds = (
+            bounds[0] - x_res * buffer,
+            bounds[1] - y_res * buffer,
+            bounds[2] + x_res * buffer,
+            bounds[3] + y_res * buffer,
+        )
+
+        # new output size
+        out_height += int(buffer * 2)
+        out_width += int(buffer * 2)
+
+        # re-calculate the transform given the new bounds, height and width
+        vrt_transform, vrt_width, vrt_height = get_vrt_transform(
+            src_dst, bounds, out_height, out_width, dst_crs=dst_crs
+        )
```

Comment:
This approach will be a bit slower because of the call to `get_vrt_transform` but it will produce better result

Reply:
Yeah I'm not sure there is a way around this because we need to run `get_vrt_transform` first before doing these buffer calculations 🤷